File: UI/src/main/java/ca/warp7/rt/java/scanner/ScannerEntry.java
package ca.warp7.rt.java.scanner;

import javafx.beans.property.BooleanProperty;
import javafx.beans.property.SimpleBooleanProperty;
import javafx.beans.property.SimpleStringProperty;
import javafx.beans.property.StringProperty;

class ScannerEntry {
    BooleanProperty commitProperty = new SimpleBooleanProperty();
    StringProperty teamProperty = new SimpleStringProperty();
    StringProperty boardScoutProperty = new SimpleStringProperty();
    StringProperty timestampProperty = new SimpleStringProperty();

    ScannerEntry(boolean commit, String team, String boardScout, String timestamp) {
        commitProperty.setValue(commit);
        teamProperty.setValue(team);
        boardScoutProperty.setValue(boardScout);
        timestampProperty.set(timestamp);
    }
}

File: UI/src/main/kotlin/ca/warp7/rt/feature/scanner/ScannerEntry.kt
package ca.warp7.rt.feature.scanner

import javafx.beans.property.BooleanProperty
import javafx.beans.property.SimpleBooleanProperty
import javafx.beans.property.SimpleStringProperty
import javafx.beans.property.StringProperty

class ScannerEntry(commit: Boolean, team: String, boardScout: String, timestamp: String) {
    var commitProperty: BooleanProperty = SimpleBooleanProperty()
    var teamProperty: StringProperty = SimpleStringProperty()
    var boardScoutProperty: StringProperty = SimpleStringProperty()
    var timestampProperty: StringProperty = SimpleStringProperty()

    init {
        commitProperty.value = commit
        teamProperty.value = team
        boardScoutProperty.value = boardScout
        timestampProperty.set(timestamp)
    }
}
